tinygrad/ops.py

```diff
@@ -203,20 +204,35 @@ def dfs(x:Any, cache:dict):
 
 class UOpMetaClass(type):
   ucache:Dict[Tuple, weakref.ReferenceType[UOp]] = {}
-  def __call__(cls, op:Ops, dtype:DType=dtypes.void, src:Tuple[UOp,...]=tuple(), arg:Any=None):
+  def __call__(cls, op:Ops, dtype:DType=dtypes.void, src:Tuple[UOp,...]=tuple(), arg:Any=None,
+               # NOTE: these don't live on the UOp class, they're internal state a pickled uop can map to
+               _device_buffer:Optional[Buffer]=None, _buf_uop:Optional[UOp]=None):
```

Comment:
I don't like this. If the UOp cache hits, this won't set. Why does this have to be done here?

Reply:
yea I need it for pickle. If I return the BUFFER in`__reduce__` I need to somehow add it back in.
I'll try to see if this can be handled in the Buffer init maybe